calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -505,3 +518,103 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
+	/*
+		if sm.GetNamespace() != cm.GetNamespace() {
+			log.Warning("Cannot change Metadata.Namespace. Reverting to past value.")
+			cm.SetNamespace(sm.GetNamespace())
+		}
+		if sm.GetName() != cm.GetName() {
+			log.Warning("Cannot change Metadata.Name. Reverting to past value.")
+			cm.SetName(sm.GetName())
+		}
+		if sm.GetGenerateName() != cm.GetGenerateName() {
+			log.Warning("Cannot change Metadata.GenerateName. Reverting to past value.")
+			cm.SetGenerateName(sm.GetGenerateName())
+		}
+		if sm.GetUID() != cm.GetUID() {
+			log.Warning("Cannot change Metadata.UID. Reverting to past value.")
+			cm.SetUID(sm.GetUID())
+		}
+		if sm.GetResourceVersion() != cm.GetResourceVersion() {
+			log.Warning("Cannot change Metadata.ResourceVersion. Reverting to past value.")
+			cm.SetResourceVersion(sm.GetResourceVersion())
+		}
+		if sm.GetGeneration() != cm.GetGeneration() {
+			log.Warning("Cannot change Metadata.Generation. Reverting to past value.")
+			cm.SetGeneration(sm.GetGeneration())
+		}
+		if sm.GetSelfLink() != cm.GetSelfLink() {
+			log.Warning("Cannot change Metadata.SelfLink. Reverting to past value.")
+			cm.SetSelfLink(sm.GetSelfLink())
+		}
+		if sm.GetCreationTimestamp() != cm.GetCreationTimestamp() {
+			log.Warning("Cannot change Metadata.CreationTimestamp. Reverting to past value.")
+			cm.SetCreationTimestamp(sm.GetCreationTimestamp())
+		}
+		if sm.GetDeletionTimestamp() != cm.GetDeletionTimestamp() {
+			log.Warning("Cannot change Metadata.DeletionTimestamp. Reverting to past value.")
+			cm.SetDeletionTimestamp(sm.GetDeletionTimestamp())
+		}
+		if sm.GetDeletionGracePeriodSeconds() != cm.GetDeletionGracePeriodSeconds() {
+			log.Warning("Cannot change Metadata.DeletionGracePeriodSeconds. Reverting to past value.")
+			cm.SetDeletionGracePeriodSeconds(sm.GetDeletionGracePeriodSeconds())
+		}
+		if sm.GetInitializers() != cm.GetInitializers() {
+			log.Warning("Cannot change Metadata.Initializers. Reverting to past value.")
+			cm.SetInitializers(sm.GetInitializers())
+		}
+		if !util.StringSlicesEqual(sm.GetFinalizers(), cm.GetFinalizers()) {
+			log.Warning("Cannot change Metadata.Finalizers. Reverting to past value.")
+			cm.SetFinalizers(sm.GetFinalizers())
+		}
+		if !util.OwnerReferenceSlicesEqual(sm.GetOwnerReferences(), cm.GetOwnerReferences()) {
+			log.Warning("Cannot change Metadata.OwnerReferences. Reverting to past value.")
+			cm.SetOwnerReferences(sm.GetOwnerReferences())
+		}
+		if sm.GetClusterName() != cm.GetClusterName() {
+			log.Warning("Cannot change Metadata.ClusterName. Reverting to past value.")
+			cm.SetClusterName(sm.GetClusterName())
+		}
+		return changed
+	*/
+}
+
+// clearInvalidMetadata removes the metadata fields that should not be set from
+// calicoctl. Any metadata fields that need to be populated except for Name, Namespace,
+// ResourceVersion, Labels, and Annotations will be set elsewhere.
+func clearInvalidMetadata(r ResourceObject) ResourceObject {
```

Comment:
Similarly, I think this method might be better off indicating that it's preparing the ObjectMetadata for a Create (I don't think this is called from anywhere else).

Reply:
Also the comment says ResourceVersion, Labels etc. are set elsewhere, but that is not true.

Reply:
The comment should say "except" for the relevant fields, but its not a very clear comment, I'll rewrite it so it is easier to read.